include/libhal-stm32f4/spi.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 Khalil Estell
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstdint>
+#include <span>
+
+#include <libhal/spi.hpp>
+
+#include "constants.hpp"
+#include "pin.hpp"
+
+namespace hal::stm32f4 {
+class spi : public hal::spi
+{
+public:
+  /// Information used to configure the spi bus
+  struct bus_info
+  {
+    /// peripheral id used to power on the spi peripheral at creation
+    peripheral peripheral_id;
+    /// spi data pin
+    pin clock;
+    /// spi clock pin
+    pin data_out;
+    /// spi clock pin
+    pin data_in;
+  };
+
+  /**
+   * @brief Construct a new spi object
+   *
+   * @param p_bus
+   * @param p_settings
+   */
+  spi(std::uint8_t p_bus, const spi::settings& p_settings = {});
```

Comment:
```suggestion
  spi(hal::runtime, std::uint8_t p_bus, const spi::settings& p_settings = {});
```

You'll need to add the `#incude <libhal/initializers.hpp>`

Reply:
This will make it clear to users that this performs runtime checks to see if the bus valid.